### DISCUSSION TITLE
Welcome to dba-dash Discussions!

### DISCUSSION THREAD

**Discussion by DavidWiseman** [trimble-oss/dba-dash#68] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

If you prefer slack, there is a #dbadash slack channel on the SQL Server Community slack.  You can join the community slack via the link below:
https://dbatools.io/slack

**Comment by kandavel88** [comment]
Hi DavidWiseaman,

We need some clarification. It should provide query suggestions like missing index  when we encounter slow-running queries or blocking, and guidance on how to configure deadlocks.

**Reply by DavidWiseman** [reply]
DBA Dash can capture slow running queries on the [Slow Queries](https://dbadash.com/docs/help/slow-queries/) tab.  Clicking on the session ID will display the associated running query snapshots for that session - showing you what the query was doing at various points in it's execution.  If plan collection is enabled, you will have access to the estimated execution plan which should include missing index suggestions for the query.  You also have the captured query along with parameters which you could re-run to capture the plan with execution stats.  Note: Missing index suggestions are quite limited and need to be evaluated carefully.  The suggestions might provide limited benefit and duplicate existing indexes.  The cost to maintain the new index is not something that is considered.

Blocking is highlighted on the performance tab.  Also in the Waits.  The Running Query snapshots provide the detailed information required to troubleshoot. If a query is blocked, you might need to focus on the tuning the query causing the blocking.  If you have a lot of blocking between readers/writers it might be worth investigating RCSI.

DBA Dash can show you how many deadlocks are occurring on the *Metrics* tab - Number of Deadlocks/sec.  For detailed analysis of deadlocks, use sp_BlitzLock.  This can also be run from DBA Dash under [Community Tools](https://dbadash.com/docs/help/community-tools/).
